### DISCUSSION TITLE
Workflow is not set as completed even though the corresponding pod is

### DISCUSSION THREAD

**Discussion by FactorizeD** [argoproj/argo-workflows#13663] [Q&A] [answered]
Hi,

To begin, I tried to search for a similar item in the discussions or issues but haven't managed to found something that would fully answer my question.

The question is actually pretty simple - when can I expect the workflow to be set as completed (in other words, having a label `workflows.argoproj.io/completed=true` (argo-workflows 3.5.4)? Unfortunately I am not an expert when it comes to argo-workflows but my expectation has always been that the workflow would be marked as completed when all pods related to its tasks would finish in some phase (doesn't matter what phase). However, it doesn't seem to be the case in situations where a given task has initially failed (pod deleted), and was then retried and finished with `Succeeded` phase.

Is that expected? In fact, I am mostly interested in the most reliable method of finding all workflow objects that have completed in a sense that all "work" related to them is already done. I need to do it using `kubectl`. What would be the best method?

Best wishes,
Mateusz

**Comment by jswxstw** [comment, marked as answer]
> However, it doesn't seem to be the case in situations where a given task has initially failed (pod deleted), and was then retried and finished with Succeeded phase.

Have you tested with v3.5.11, this issue should have been fixed by #13537.

**Reply by FactorizeD** [reply]
Ah, haven't seen this one. Thanks a lot for a quick response!
